### DISCUSSION TITLE
How custom devices are matched when adding a devices in localtuya HA ?

### DISCUSSION THREAD

**Discussion by davidchezmoa** [make-all/tuya-local#2696] [General]
Hi,

When I add a device to my localtuya by IP, ID and localkey, a popup comes and propose some device config that could match ? but how the matching is done ?

In my particualar case I have a "tonepie cat litter box" to add in localtuya and I know this devices is supported but when I add it to localtuya, it propose me a "garage door opener" and no way to manually select an other thing.

What I need to do ?

Thank you.

**Comment by make-all** [comment]
Off topic

**Comment by davidchezmoa** [comment]
Wonderfull
